### PR TITLE
Improve search for video files

### PR DESCRIPTION
**Pull request by ihhub**
The same technique as for music files: we put directory where we find a current video file at the first place so for the subsequent runs we will search there.
Also avoid copying of strings where it is not necessary.

### CODE REVIEW

File: src/fheroes2/game/game_video.cpp [hunk 1]
```diff
@@ -31,10 +31,12 @@
 #include "tools.h"
 #include "ui_tool.h"
 
+#include <array>
+
 namespace
 {
     // Anim2 directory is used in Russian Buka version of the game.
-    const std::vector<std::string> videoDir = { "anim", "anim2", System::ConcatePath( "heroes2", "anim" ), "data" };
+    std::array<std::string, 4> videoDir = { "anim", "anim2", System::ConcatePath( "heroes2", "anim" ), "data" };
```
Comment (oleg-derevenetz) [inline src/fheroes2/game/game_video.cpp:39]: Just curious, where is the `heroes2/anim` variant used?

**Review by oleg-derevenetz — APPROVED** [review]

File: src/fheroes2/game/game_video.cpp [hunk 2]
```diff
@@ -63,9 +65,16 @@ namespace Video
                     std::string targetFileName = System::ConcatePath( fullDirPath, fileName );
```
Comment (oleg-derevenetz) [inline src/fheroes2/game/game_video.cpp:65]: @ihhub I wonder, should we do all this? It seems that `FindFileInDir()` should do all the job, so if the `videoFiles` contains something, then it already should be what we need.

File: src/fheroes2/game/game_video.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/game/game_video.cpp:39]: In Windows GoG version of the game.

File: src/fheroes2/game/game_video.cpp (hunk 2 above)
Comment (ihhub) [reply src/fheroes2/game/game_video.cpp:65]: `FindFileInDir` search all files by postfix. For example, `const std::string pattern( nameAsFilter ? path + "\\*" + name : path + "\\" + name );`. In this case we either need to change the logic within `getFilesFromDirectory` function or keep the current logic as it is.

Comment (oleg-derevenetz) [reply src/fheroes2/game/game_video.cpp:65]: @ihhub But `FindFileInDir()` calls the `getFilesFromDirectory()` with `nameAsFilter = false`, i.e. exact match is required. This is its difference from the `ReadDir()` which search by postfix by calling the `getFilesFromDirectory()` with `nameAsFilter = true` (BTW this is in fact the only difference between `FindFileInDir()` and `ReadDir()`).

File: src/fheroes2/game/game_video.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/fheroes2/game/game_video.cpp:39]: @ihhub Hmm, I always thought that GoG version doesn't have `ANIM` directory at all, so we use the special script to extract animations from the GOG CD image. Or is it not always so?

Comment (ihhub) [reply src/fheroes2/game/game_video.cpp:39]: This is for Windows version of the game. DOS version does not have such directory. Also users can manually copy files.

Comment (oleg-derevenetz) [reply src/fheroes2/game/game_video.cpp:39]: Hi @ihhub can you please post here the file layout for GOG Windows version (like `ls -lR` output)? I need to check if the importing script properly supports it.

Comment (ihhub) [reply src/fheroes2/game/game_video.cpp:39]: It is on my Windows:
```
d:\Heroes of Might and Magic 2 (Win)>dir /s /b /o:gn
d:\Heroes of Might and Magic 2 (Win)\DATA
d:\Heroes of Might and Magic 2 (Win)\EULA
d:\Heroes of Might and Magic 2 (Win)\GAMES
d:\Heroes of Might and Magic 2 (Win)\HELP
d:\Heroes of Might and Magic 2 (Win)\HEROES2
d:\Heroes of Might and Magic 2 (Win)\Journals
d:\Heroes of Might and Magic 2 (Win)\MAPS
d:\Heroes of Might and Magic 2 (Win)\MUSIC
d:\Heroes of Might and Magic 2 (Win)\EDITOR2W.EXE
d:\Heroes of Might and Magic 2 (Win)\EULA.txt
d:\Heroes of Might and Magic 2 (Win)\GameuxInstallHelper.dll
d:\Heroes of Might and Magic 2 (Win)\gfw_high.ico
d:\Heroes of Might and Magic 2 (Win)\gog.ico
d:\Heroes of Might and Magic 2 (Win)\gog_homm2_win8.sdb
d:\Heroes of Might and Magic 2 (Win)\goggame.dll
d:\Heroes of Might and Magic 2 (Win)\Graphic Setup.exe
d:\Heroes of Might and Magic 2 (Win)\H2CAMP.TXT
d:\Heroes of Might and Magic 2 (Win)\HEROES2W.EXE
d:\Heroes of Might and Magic 2 (Win)\INSTALL.EXE
d:\Heroes of Might and Magic 2 (Win)\Launch Heroes of Might and Magic 2 GOLD.lnk
d:\Heroes of Might and Magic 2 (Win)\libogg-0.dll
d:\Heroes of Might and Magic 2 (Win)\libvorbis-0.dll
d:\Heroes of Might and Magic 2 (Win)\libvorbisfile-3.dll
d:\Heroes of Might and Magic 2 (Win)\manual.pdf
d:\Heroes of Might and Magic 2 (Win)\MSS32.DLL
d:\Heroes of Might and Magic 2 (Win)\POLCAMP.TXT
d:\Heroes of Might and Magic 2 (Win)\README.TXT
d:\Heroes of Might and Magic 2 (Win)\SMACKW32.DLL
d:\Heroes of Might and Magic 2 (Win)\Support.ico
d:\Heroes of Might and Magic 2 (Win)\unins000.dat
d:\Heroes of Might and Magic 2 (Win)\unins000.exe
d:\Heroes of Might and Magic 2 (Win)\unins000.ini
d:\Heroes of Might and Magic 2 (Win)\unins000.msg
d:\Heroes of Might and Magic 2 (Win)\winmm.dll
d:\Heroes of Might and Magic 2 (Win)\DATA\CAMPAIGN.HS
d:\Heroes of Might and Magic 2 (Win)\DATA\HEROES2.AGG
d:\Heroes of Might and Magic 2 (Win)\DATA\HEROES2X.AGG
d:\Heroes of Might and Magic 2 (Win)\DATA\IVYDES.SMK
d:\Heroes of Might and Magic 2 (Win)\DATA\IVYPOL.SMK
d:\Heroes of Might and Magic 2 (Win)\DATA\IVYVOY.SMK
d:\Heroes of Might and Magic 2 (Win)\DATA\IVYWIZ.SMK
d:\Heroes of Might and Magic 2 (Win)\DATA\NETLODR.DAT
d:\Heroes of Might and Magic 2 (Win)\DATA\STANDARD.HS
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT CAN English--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT CAN French--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT Chinese--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT CS--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT DEN--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT Export teritories--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT Finish--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT France + Canada (FR)--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT France--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT GER--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT HU--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT IT--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT NLD--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT Norwegian--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT PL--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT RU--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT SPA--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT SWE--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT UK--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\EULA\EULA UBISOFT US--Jan09.doc
d:\Heroes of Might and Magic 2 (Win)\GAMES\AUTOSAVE.GM1
d:\Heroes of Might and Magic 2 (Win)\GAMES\AUTOSAVE.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\AUTOSAVE.GX1
d:\Heroes of Might and Magic 2 (Win)\GAMES\AUTOSAVE.GXC
d:\Heroes of Might and Magic 2 (Win)\GAMES\Near Jail 01.GX1
d:\Heroes of Might and Magic 2 (Win)\GAMES\Passability 14.GX1
d:\Heroes of Might and Magic 2 (Win)\GAMES\Passability near Mermaid.GX1
d:\Heroes of Might and Magic 2 (Win)\GAMES\TUTORIAL.GM1
d:\Heroes of Might and Magic 2 (Win)\GAMES\VOY_1.GXC
d:\Heroes of Might and Magic 2 (Win)\GAMES\VOY_2.GXC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_E_01.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_E_02.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_E_03.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_E_04.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_E_05.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_E_06.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_E_08.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_E_09.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_G_01.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_G_02.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_G_03.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_G_04.GMC
d:\Heroes of Might and Magic 2 (Win)\GAMES\WIN_G_05.GMC
d:\Heroes of Might and Magic 2 (Win)\HELP\HEROES2.CNT
d:\Heroes of Might and Magic 2 (Win)\HELP\HEROES2.HLP
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\CHOOSE.SMK
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\CHOOSEW.SMK
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\CREDITS.SMK
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\CYBCREDS.SMK
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\cylogo.smk
... <---- here I purposely removed lines as no need to show all video files
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\WIN.SMK
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\WIZ16.SMK
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\WIZ17.SMK
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\WIZ18.SMK
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\WIZ19.SMK
d:\Heroes of Might and Magic 2 (Win)\HEROES2\ANIM\WIZ20.SMK
d:\Heroes of Might and Magic 2 (Win)\Journals\betrayal.txt
d:\Heroes of Might and Magic 2 (Win)\Journals\GHOST.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\GOHNOTES.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\ROADHOME.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\ROCNOTES.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\THUNK1.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\THUNK2.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\THUNK3.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\THUNK4.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\THUNK5.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\WIZARD1.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\WIZARD1A.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\WIZARD2.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\WIZARD2A.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\WIZARD3.TXT
d:\Heroes of Might and Magic 2 (Win)\Journals\WIZARD3A.TXT
d:\Heroes of Might and Magic 2 (Win)\MAPS\2kings.MX2
d:\Heroes of Might and Magic 2 (Win)\MAPS\Alteris.mp2
d:\Heroes of Might and Magic 2 (Win)\MAPS\Alteris2.mp2
d:\Heroes of Might and Magic 2 (Win)\MAPS\Animages.mp2
d:\Heroes of Might and Magic 2 (Win)\MAPS\ARRAX.MX2
d:\Heroes of Might and Magic 2 (Win)\MAPS\BELTWAY.MP2
d:\Heroes of Might and Magic 2 (Win)\MAPS\Betrayal.mp2
d:\Heroes of Might and Magic 2 (Win)\MAPS\BlackFor.mp2
d:\Heroes of Might and Magic 2 (Win)\MAPS\BROKENA.MP2
... <---- here I purposely removed lines as no need to show all maps
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track02.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track03.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track04.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track05.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track06.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track07.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track08.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track09.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track10.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track11.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track12.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track13.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track14.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track15.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track16.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track17.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track18.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track19.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track20.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track21.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track22.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track23.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track24.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track25.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track26.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track27.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track28.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track29.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track30.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track31.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track32.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track33.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track34.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track35.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track36.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track37.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track38.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track39.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track40.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track41.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track42.ogg
d:\Heroes of Might and Magic 2 (Win)\MUSIC\Track43.ogg
```

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**